eclipse-cdt-cloud/vscode-memory-inspector
# Context menu enhancements

#### What it does
Enhances the memory webview to  properly support context menu actions contributed via `webview/context` contribution point. This includes
- Augmenting the webview components with `data-vscode-context` custom data properties.
 These properties are used to provide additional context info (as JSON string). The composed context is then available in `when` conditions from `webview/context` contributions and as command argument when executing the associated command.

Provides the following context menu enhancements outlined in #51
- Copy selection: Copies the current text selection/datatable selection to the clipboard. Text selection has priority.
   If no text is selected the value of the selected datatable cell will be copied instead.
- Quick access to window configuration by
  - providing show/hide entries for variable & ascii column
  - show/hide for radix prefix
  - `Show advanced Options` command to show the advanced settings overlay
 - Allow contributions from other extensions


#### How to test

Open a context menu in the memory webview. The display entries (Show/Hide xyz, Show Advanced Options) should always be visible and work as expected.

Open a context menu ontop of a variable entry in the `variables` column. The `Test variable context` entry should appear.
On click it will print the following to the memory output channel:
- Current webview selection (retrieved via the newly introduced command)
- The complete context that was applied when opening the context menu

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
